### PR TITLE
BaSP-MR-25: get, getbyid and create on activities

### PR DESCRIPTION
**Pull request by Roberto-Orazi**
In this branch i do the test for the activities feature.

### CODE REVIEW

File: src/tests/activities.test.js [hunk 1]
```diff
@@ -1,5 +1,54 @@
...
+});
+afterEach(async () => {
+  await Activity.collection.deleteMany();
+});
+
+describe('GET /api/activities', () => {
+  test('It should respond with an array of activities', async () => {
+    const response = await request(app).get(`/api/activities/${activityId}`).send();
+    expect(response.body).toEqual(expect.arrayContaining(activitySeed));
+    expect(response.statusCode).toBe(200);
```
Comment (JuanManuel-Blois) [inline src/tests/activities.test.js:24]: we should check the error, message and data in the test

File: src/tests/activities.test.js [hunk 2]
```diff
@@ -1,5 +1,54 @@
...
+  });
+  test('It should respond with a 404', async () => {
+    const response = await request(app).get('/api/activities/');
+    expect(response.statusCode).toBe(404);
```
Comment (JuanManuel-Blois) [inline src/tests/activities.test.js:28]: we should check the error, message and data in the test

File: src/tests/activities.test.js [hunk 3]
```diff
@@ -1,5 +1,54 @@
...
+  });
+});
+
+describe('POST /api/activities', () => {
+  test('It should respond with a new activity', async () => {
+    const response = await request(app).post(`/api/activities/${activityId}`).send(mockActivity);
+    expect(response.body).toMatchObject(mockActivity);
+    expect(response.statusCode).toBe(200);
```
Comment (JuanManuel-Blois) [inline src/tests/activities.test.js:36]: we should check the error, message and data in the test

File: src/tests/activities.test.js [hunk 4]
```diff
@@ -1,5 +1,54 @@
...
+    expect(response.statusCode).toBe(404);
+  });
+});
+
+describe('GET /api/activities/:id', () => {
+  test('It should respond with a single activity', async () => {
+    const response = await request(app).get('/api/activities/1');
+    expect(response.body).toMatchObject(activitySeed[0]);
+    expect(response.statusCode).toBe(200);
+  });
```
Comment (JuanManuel-Blois) [inline src/tests/activities.test.js:49]: we should check the error, message and data in the test

File: src/tests/activities.test.js [hunk 5]
```diff
@@ -1,5 +1,54 @@
...
+  test('It should respond with a 404', async () => {
+    const response = await request(app).post(`/api/activities/${activityId}`).send({});
+    expect(response.statusCode).toBe(404);
+  });
```
Comment (JuanManuel-Blois) [inline src/tests/activities.test.js:53]: we should check the error, message and data in the test

File: src/tests/activities.test.js [hunk 6]
```diff
@@ -1,5 +1,54 @@
...
+beforeEach(async () => {
+  await Activity.collection.insertMany(activitySeed);
+});
+afterEach(async () => {
+  await Activity.collection.deleteMany();
+});
+
+describe('GET /api/activities', () => {
+  test('It should respond with an array of activities', async () => {
+    const response = await request(app).get(`/api/activities/${activityId}`).send();
```
Comment (JuanManuel-Blois) [inline src/tests/activities.test.js:22]: this is calling an specific activity by id, the get method should bring all the activities

File: jest.setup.js [hunk 7]
```diff
@@ -22,4 +22,4 @@ afterAll(async () => {
   await mongoose.connection.close();
   await mongoServer.stop();
   await mongo.stop();
-});
+}, 30000);
```
Comment (JuanManuel-Blois) [inline jest.setup.js:25]: change this back to 10000

**Review by LucianoAlarcon — APPROVED** [review]

**Review by nicolasblasco — APPROVED** [review]

**Review by JuanManuel-Blois — APPROVED** [review]

**Pull request merged**